src/improvements/template/EnableDeputyPauseModuleTemplate.sol

```diff
@@ -0,0 +1,188 @@
+    }
+
+    /// @notice No code exceptions for this template
+    function getCodeExceptions() internal view override returns (address[] memory) {}
+
+    /// @notice helper function that can be overridden by template contracts to
+    /// check the state changes applied by the task. This function can check
+    /// that only the nonce changed in the parent multisig when executing a task
+    /// by checking the slot and address where the slot changed.
+    function checkStateDiff(VmSafe.AccountAccess[] memory accountAccesses) internal override {
+        super.checkStateDiff(accountAccesses);
+
+        (address[] memory modules,) = ModuleManager(parentMultisig).getModulesPaginated(SENTINEL_MODULE, 100);
+
+        bytes32 moduleSlot = keccak256(abi.encode(newModule, MODULE_MAPPING_STORAGE_OFFSET));
+        bytes32 sentinelSlot = keccak256(abi.encode(SENTINEL_MODULE, MODULE_MAPPING_STORAGE_OFFSET));
+
+        bool moduleWriteFound;
+
+        for (uint256 i; i < accountAccesses.length; i++) {
+            VmSafe.AccountAccess memory accountAccess = accountAccesses[i];
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                VmSafe.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+                if (!storageAccess.isWrite) continue; // Skip SLOADs.
+                if (storageAccess.isWrite && storageAccess.account == parentMultisig) {
+                    assertTrue(
+                        storageAccess.slot == NONCE_STORAGE_OFFSET || storageAccess.slot == moduleSlot
+                            || storageAccess.slot == sentinelSlot,
+                        "Only nonce and module slot should be updated on upgrade controller multisig"
+                    );
+
+                    if (storageAccess.slot == moduleSlot) {
+                        assertEq(
+                            address(uint160(uint256(storageAccess.newValue))),
+                            modules.length >= 2 ? modules[1] : SENTINEL_MODULE,
+                            "new module not correct"
+                        );
+
+                        bytes32 sentinelModuleValue = vm.load(parentMultisig, sentinelSlot);
+                        assertEq(
+                            sentinelModuleValue,
+                            bytes32(uint256(uint160(newModule))),
+                            "sentinel does not point to new module"
+                        );
+
+                        moduleWriteFound = true;
+                    }
+                }
+            }
+        }
+
+        /// module write must be found, else revert
+        assertTrue(moduleWriteFound, "Module write not found");
+    }
+}
+
+interface IDeputyGuardianModuleFetcher {
+    function deputyGuardian() external view returns (address);
+    function foundationSafe() external view returns (address);
+    function superchainConfig() external view returns (address);
+    function version() external view returns (string memory);
+}
```

Comment:
We can delete this and just import the actual interface from `lib/optimism/packages/contracts-bedrock/interfaces/safe/IDeputyPauseModule.sol`

Reply:
Resolving.